inc/boost/module.modulemap

```diff
@@ -2324,18 +2324,26 @@ module boost_graph_pending_random_wrapper {
     module "pending__detail__property" { header "pending/detail/property.hpp" export * }
     module "pending__disjoint_sets" { header "pending/disjoint_sets.hpp" export * }
     module "pending__fenced_priority_queue" { header "pending/fenced_priority_queue.hpp" export * }
-    module "pending__fibonacci_heap" { header "pending/fibonacci_heap.hpp" export * }
-    module "pending__indirect_cmp" { header "pending/indirect_cmp.hpp" export * }
     module "pending__integer_log2" { header "pending/integer_log2.hpp" export * }
     module "pending__is_heap" { header "pending/is_heap.hpp" export * }
     module "pending__iterator_adaptors" { header "pending/iterator_adaptors.hpp" export * }
     module "pending__lowest_bit" { header "pending/lowest_bit.hpp" export * }
-    module "pending__mutable_queue" { header "pending/mutable_queue.hpp" export * }
     module "pending__property" { header "pending/property.hpp" export * }
     module "pending__property_serialize" { header "pending/property_serialize.hpp" export * }
     module "pending__queue" { header "pending/queue.hpp" export * }
-    module "pending__relaxed_heap" { header "pending/relaxed_heap.hpp" export * }
     module "pending__stringtok" { header "pending/stringtok.hpp" export * }
+  /*
+  All these modules caused this "declaration of 'if_'.." error, so removed them from modulemap
+/home/travis/build/Teemperor/boost-compile/inc/boost/property_map/property_map.hpp:60:17: error:
+      declaration of 'if_' must be imported from module 'boost_mpl.mpl__bind'
+      before it is required
+    boost::mpl::if_<is_property_map<PA>,
```

Comment:
Would that mean that we are missing a `#include <mpl/bind>`? Maybe it is worth in submitting a PR with adding this header file.

Reply:
This quick check doesn't seem to confirm this, so it's a modules bug:
```diff
diff --git a/inc/boost/property_map/property_map.hpp b/inc/boost/property_map/property_map.hpp
index cd6dd55..b822602 100644
--- a/inc/boost/property_map/property_map.hpp
+++ b/inc/boost/property_map/property_map.hpp
@@ -55,6 +55,9 @@ namespace boost {
  
   struct null_property_traits {};
  
+  #ifndef BOOST_MPL_IF_HPP_INCLUDED
+   #error "Foo"
+  #endif
   template <typename PA>
   struct property_traits :
     boost::mpl::if_<is_property_map<PA>,
```

But directly including the header from this file fixes it, so somehow this decl isn't propagated when implicitly included by something else.

Reply:
Ok, figured it out.